asset/cloud-client/quickstart_searchallresources_test.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import uuid
+
+import backoff
+from google.api_core.exceptions import InvalidArgument, NotFound
+from google.cloud import bigquery
+import pytest
+
+import quickstart_searchallresources
+
+PROJECT = os.environ['GOOGLE_CLOUD_PROJECT']
+DATASET = 'dataset_{}'.format(uuid.uuid4().hex)
+
+
+@pytest.fixture(scope='module')
+def bigquery_client():
+    yield bigquery.Client()
+
+
+@pytest.fixture(scope='module')
+def asset_dataset(bigquery_client):
+    dataset = bigquery_client.create_dataset(DATASET)
+
+    yield DATASET
+
+    try:
+        bigquery_client.delete_dataset(dataset)
+    except NotFound as e:
+        print('Failed to delete dataset {}'.format(DATASET))
+        raise e
+
+
+def test_search_all_resources(asset_dataset, capsys):
+    scope = "projects/{}".format(PROJECT)
+    query = "name:{}".format(DATASET)
+
+    @backoff.on_exception(
+        backoff.expo, (AssertionError, InvalidArgument), max_time=30
```

Comment:
Why are we added a backoff for this sample? Can we add a comment with context on why this is?

Reply:
It takes some time for the created dataset to be searchable. I realized sleep() may be more intuitive in this case than using backoff to retry the search.

Reply:
Sorry, but why? I recommend backoff over sleep.

Reply:
Ok I'll switch back to backoff. I don't code python a lot so may I ask why backoff is preferred?

Reply:
Using backoff is tmatsuo's personal preference, and it's not one I agree with here. I believe `backoff.on_exception` has max_tries default to `None`, which means this will never fail and instead will continue to retry until the whole build fails out.

Reply:
I have `max_time` set to 30. It take no more than a few seconds to propagate, so maybe reduce it to 10s?

Reply:
I'm fine with 30 - the difference seems negligible.